crates/core/tedge_agent/src/software_list_manager/actor.rs

```diff
@@ -0,0 +1,164 @@
+use crate::software_list_manager::config::SoftwareListManagerConfig;
+use crate::software_list_manager::error::SoftwareListManagerError;
+use crate::software_list_manager::error::SoftwareListManagerError::NoPlugins;
+use crate::state_repository::state::AgentStateRepository;
+use crate::state_repository::state::SoftwareOperationVariants;
+use crate::state_repository::state::State;
+use crate::state_repository::state::StateRepository;
+use crate::state_repository::state::StateStatus;
+use async_trait::async_trait;
+use plugin_sm::operation_logs::LogKind;
+use plugin_sm::operation_logs::OperationLogs;
+use plugin_sm::plugin_manager::ExternalPlugins;
+use std::sync::Arc;
+use tedge_actors::Actor;
+use tedge_actors::MessageReceiver;
+use tedge_actors::RuntimeError;
+use tedge_actors::Sender;
+use tedge_actors::SimpleMessageBox;
+use tedge_api::OperationStatus;
+use tedge_api::SoftwareListRequest;
+use tedge_api::SoftwareListResponse;
+use tedge_api::SoftwareRequestResponse;
+use tokio::sync::Mutex;
+use tracing::error;
+use tracing::log::warn;
+
+#[cfg(not(test))]
+const SUDO: &str = "sudo";
+#[cfg(test)]
+const SUDO: &str = "echo";
+
+pub struct SoftwareListManagerActor {
+    config: SoftwareListManagerConfig,
+    state_repository: AgentStateRepository,
+    operation_logs: OperationLogs,
+    message_box: SimpleMessageBox<SoftwareListRequest, SoftwareListResponse>,
+}
+
+#[async_trait]
+impl Actor for SoftwareListManagerActor {
+    fn name(&self) -> &str {
+        "SoftwareListManagerActor"
+    }
+
+    async fn run(&mut self) -> Result<(), RuntimeError> {
+        let plugins = Arc::new(Mutex::new(
```

Comment:
Using an `Arc<Mutex< >>` is useless here as the `plugins` set is never used concurrently.